src/_pytest/nodes.py

```diff
@@ -697,6 +684,34 @@ def __init__(
         #: for this test.
         self.user_properties: List[Tuple[str, object]] = []
 
+        self._check_item_and_collector_diamond_inheritance()
+
+    # Set of node types to avoid issuing a diamond inheritance warning
+    # for the same class more than once, which is not helpful.
+    _DIAMOND_COLLECTOR_WARNING: ClassVar[Set[Type["Item"]]] = set()
```

Comment:
Not sure we want this, might be better to just let we issue multiple warnings and let users filter them. Thoughts?

Reply:
Any thoughts on this particular aspect? @RonnyPfannschmidt @The-Compiler ?

Reply:
"classically" warnings once filters should sort that, however as we cant rely on them like wed like, i think its fine to do this hack

Reply:
@nicoddemus we might however use a boolean on each individual type instance instead of the set in the base class